packages/next-swc/crates/core/src/emotion/mod.rs

```diff
@@ -0,0 +1,758 @@
+                },
+                ExportItem {
+                    name: "keyframes".to_owned(),
+                    kind: ExprKind::Css,
+                },
+                ExportItem {
+                    name: "Global".to_owned(),
+                    kind: ExprKind::GlobalJSX,
+                },
+            ],
+            ..Default::default()
+        },
+        EmotionModuleConfig {
+            module_name: "@emotion/primitives".into(),
+            exported_names: vec![ExportItem {
+                name: "css".to_owned(),
+                kind: ExprKind::Css,
+            }],
+            default_export: Some(ExprKind::Styled),
+        },
+        EmotionModuleConfig {
+            module_name: "@emotion/native".into(),
+            exported_names: vec![ExportItem {
+                name: "css".to_owned(),
+                kind: ExprKind::Css,
+            }],
+            default_export: Some(ExprKind::Styled),
+        },
+    ]
+});
+
+static SPACE_AROUND_COLON: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"\s*(?P<s>[:|;|,|\{,\}])\s*").unwrap());
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct EmotionOptions {
+    pub enabled: Option<bool>,
+    pub sourcemap: Option<bool>,
+    pub auto_label: Option<bool>,
+    pub label_format: Option<String>,
+}
+
+impl Default for EmotionOptions {
+    fn default() -> Self {
+        EmotionOptions {
+            enabled: Some(false),
+            sourcemap: Some(true),
+            auto_label: Some(true),
+            label_format: Some("[local]".to_owned()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct EmotionModuleConfig {
+    module_name: JsWord,
+    exported_names: Vec<ExportItem>,
+    default_export: Option<ExprKind>,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+struct ExportItem {
+    name: String,
+    kind: ExprKind,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+enum ImportType {
+    Named,
+    Namespace,
+    Default,
+}
+
+impl Default for ImportType {
+    fn default() -> Self {
+        ImportType::Named
+    }
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+enum ExprKind {
+    Css,
+    Styled,
+    GlobalJSX,
+}
+
+impl Default for ExprKind {
+    fn default() -> Self {
+        ExprKind::Css
+    }
+}
+
+#[derive(Debug)]
+enum PackageMeta {
+    Named(ExprKind),
+    Namespace(EmotionModuleConfig),
+}
+
+pub fn emotion<C: Comments>(
+    emotion_options: EmotionOptions,
+    path: &Path,
+    cm: Arc<SourceMap>,
+    comments: C,
+) -> impl Fold {
+    EmotionTransformer::new(emotion_options, path, cm, comments)
+}
+
+pub struct EmotionTransformer<C: Comments> {
+    pub options: EmotionOptions,
+    filepath_hash: Option<u32>,
+    filepath: PathBuf,
+    dir: Option<String>,
+    filename: Option<String>,
+    cm: Arc<SourceMap>,
+    comments: C,
+    import_packages: FxHashMap<JsWord, PackageMeta>,
```

Comment:
I missed this.

Reply:
It should be `FxHashMap<Id, PackageMeta>` instead, where `Id` is `swc_ecma_ast::Id`.

Reply:
You can read doc at https://docs.rs/swc_ecma_ast/0.69.1/swc_ecma_ast/struct.Ident.html for more details.

Reply:
If we only check `symbol`, all identifiers with the same symbol will be treated as imported ones.